packages/utilities/label.tsx

```diff
@@ -42,7 +42,12 @@ export const renderLabel = ({
   const hasError = appearance === InputAppearance.Error;
   const labelClassName = hidden ? cx(visuallyHidden) : getLabelStyle(hasError);
   const labelNode = (
-    <label className={labelClassName} htmlFor={id} data-cy="textInput-label">
+    <label
+      className={labelClassName}
+      htmlFor={id}
+      data-cy="textInput-label"
+      hidden={hidden}
```

Comment:
Similar question on why the `hidden` attribute needed to be added?

Reply:
@nataliepina according to the current implementation, when we want to hide the label, we use CSS 
```
clip: rect(0 0 0 0)
```
Due to this, elements are still present in the DOM structure and we cannot test 
```
expect(queriedElement).not.toBeInTheDocument();
```
Also, we cannot use 
```
expect(queriedElement).not.toBeVisible();
```
Because according to the react testing library documentation https://github.com/testing-library/jest-dom#tobevisible

> An element is visible if all the following conditions are met:
> 
> - it is present in the document
> - it does not have its css property display set to none
> - it does not have its css property visibility set to either hidden or collapse
> - it does not have its css property opacity set to 0
> - its parent element is also visible (and so on up to the top of the DOM tree)
> - it does not have the [hidden](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden) attribute
> - if <details /> it has the open attribute

And according to MDN docs hidden attribute is used when we want to hide a specific part of the DOM without removing it from DOM. So, expecting that is what required here added this hidden flag.

Reply:
`visuallyHidden` is used for accessibility purposes, so I fear this may break that existing behavior for screen readers.

> Note on hiding elements
Screen readers and other assistive technology, just as web browsers, hide elements from their users when they are styled using display: none; and visibility: hidden;.

The common approach used to hide information visually but keep them available to screen reader and other assistive technology users is to use CSS that will keep the information technically visible but practically hidden. For example, presenting the label in a 1-by-1 pixel area, as demonstrated with the CSS class visuallyhidden (do not confuse with CSS visibility: hidden) below:

Code Snippet
.visuallyhidden {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}
Note that interactive elements are still active when using this code, so it is not suitable to hide other parts of forms but the labels.

[Source](https://www.w3.org/WAI/tutorials/forms/labels/)

Reply:
@nataliepina what do you use for accessibility testing? Chrome VOX, NVDIA or Apple's inbuild voiceover?

Reply:
I have used Apple's voiceover personally in our product app that uses UI Kit. There should be some built-in accessibility checks in Storybook with the `@storybook/addon-a11y`. You can see the Accessibility tab when the story is running. I don't see any violations here.

I saw this on MDN so I wanted to be sure:
> The hidden attribute must not be used to hide content just from one presentation. If something is marked hidden, it is hidden from all presentations, including, for instance, screen readers.

I am unfamiliar with how the hidden attribute works with a11y so I wanted to see if you had any thoughts there. After some additional reading I've done I think it should be fine to include this.